### ISSUE TITLE
The color theme is not preserved after restarts

### ISSUE THREAD

**Issue by SvetlinZarev** [eclipse-color-theme/eclipse-color-theme#223]
Hi, 

After each restart of Eclipse (Luna SR2/Mars M7) I have to setup my color theme preferences again. 
Although I can see that my preferred color theme to be selected, it is not applied. So I just have to open the menu and press apply, but it;s very annoying because it has to be done on each restart and for each workspace.

Kind regards,
Svetlin

**Comment by rleeson** [comment]
Hello,

I am also experiencing this issue, the perspective does not appear to apply the theme on load, instead the standard selected Eclipse theme applies.  My current installation is Mars 4.5.0 (20150621-1200) 64-bit with Eclipse Color Theme, IDE for PHP Developers, Git Team Provider, and Emmet for Eclipse.

This project was initializing the color theme properly on load when I was using Juno.  After installeing Mars and upgrading the project, the on load issue started.  To troubleshoot, I've removed the plugins and reinstalled fresh, and removed the preferences file.  

The preferences file is created properly, and I confirm if I switch to a different perspective (Git in this case) then back to the PHP perspective, the saved color theme loads.

I think there may be an order of execution issue, or missing event, specific to Mars (perhaps earlier versions, though as mentioned, it worked for Juno).  Let me know if I can provide more information.

Thanks,
Ryan

**Comment by zulus** [comment]
This problem is related to early oomph version (included and enabled by default in all Mars milestones).

See also bug: https://bugs.eclipse.org/bugs/show_bug.cgi?id=470430

**Comment by SvetlinZarev** [comment]
Well, that;s a workaround, not  a solution. The plugin should integrate with Oomph - i.e. it should tell Oomph to save the changes made by it.

**Comment by Catscratch** [comment]
Tried the workaroud without success. I deleted the dir in workspace, eclipse configurations and users home dir. But nothing worked. The dir is re-created again and the theme falls back to standard. I also completely removed the oomph feature from eclipse...no success.

I'm using 4.5.0 on Win8.1.

**Comment by rleeson** [comment]
The workaround to remove the Oomph settings was also unsuccessful for me.

**Comment by fschmitt** [comment]
I've encountered the same problem. Now, i've found a fix: Oomph needed a setup, running some sort of questionnaire. This setup has to be started manually when upgrading an existing workspace from an older eclipse version. To do so:

- Open Menu `Window`-> `Preferences`
- Select `Oomph` -> `Setup Tasks`
- hit button `Start Welcome Questionnaire`.
- A questionnaire dialogue will open, asking some setup questions. Take care of the sixth question: "Enable preference recorder?" - make sure to enable it:
    
![bildschirmfoto vom 2015-07-06 11 39 57](https://cloud.githubusercontent.com/assets/417262/8519530/9f1465e0-23d3-11e5-8f1c-4d85eb0230cc.png)

- After saving the answers, restarting eclipse and selecting one of the Color plugin themes, Oomph will offer to save the custom theme settings.
- Confirm saving the preferences.

That's it - now, the selected theme should be preserved after restarts.

**Comment by apolci** [comment]
I have the same problem with Spring Tools Suite 3.7.0 based on Eclipse Mars 4.5 GA and I cannot find the preference page mentioned by fschmitt.

**Comment by ibsavage** [comment]
Same problem here on Eclipse Mars. Fschmitt's solution seemed to work but didn't actually activate the preference recorder. It required me going back into the Oomph preferences and clicking the Enabled checkbox for the preference recorder before it finally worked.

**Comment by ewoks** [comment]
As others said, it doesn't work. What @fschmitt proposed has no effect at least with newest official Mars.. Annoying

**Comment by fschmitt** [comment]
@ewoks  Does it work when enabling the "preferences recorder" manually as described by @ibsavage ?

**Comment by Catscratch** [comment]
Thanks @fschmitt - worked for me now. Had to enable recorder manually.

**Comment by mondjunge** [comment]
Me also suffering from this bug. ubuntu 15.04, Eclipse Mars JEE (20150621-1200)

I got this workaround:
1. Install Color-Themes,
2. Install Jeeeyul's Eclipse Themes
3. restart
4. choose and apply or ok a color theme
5. With the Theme Button from Jeeeyul's Eclipse Themes, choose exactly the UI Theme that is currently active. (for me this is Other Themes > Dark)
6. Beautiful: The editor creates little artifacts for a moment.
7. restart. 
Outcome: I have the same editor color theme applied than selected before. Hooray.

PS: The Oomph trick @fschmitt mentioned did not work for me (yes, I also had to activate recorder manually, like @ibsavage described, and no, not even a reset of color theme, after enabling recorder like @fschmitt mentioned, worked for me.)

**Comment by vfedoriv** [comment]
@fschmitt, thanks.

**Comment by mondjunge** [comment]
There is a Mars Update which lets you (Oomph)-record your settings to the current workspace only. Now with Oomph Recording active and set to save changes in the workspace, it successfully restores my settings when starting a workspace, even the chosen editor color theme.

**Comment by krohani** [comment]
Yes it does work on the latest Mars 4.5.1 Release. As others have said you just have to manually enable the recorder.

**Comment by d00001** [comment]
I'm experiencing the same issue on Win10/Neon.

**Comment by kojotak** [comment]
I'm dealing with the same issue in STS 3.7.3.RELEASE based on Mars 4.5.2 (com.github.eclipsecolortheme.feature_1.0.0.201410260308).

**Comment by 4gatepylon** [comment]
I have dealt with this problem on neon, but fixed it by trying the steps enumerated by fschmitt manually.  You must go to window > preferences > oomph > setup tasks > preferences recorder and check the box labeled "record into."  It will prompt you whether you want to keep it locally or remotely (you would have to login for the remote option).  If you do it locally, you can decide where to keep it: by default either User, Workspace, or Installation.  Once you do this you should apply the theme you want (click ok), and restart eclispe.  Then it should load fine.

**Comment by powergold1** [comment]
For me, I also had to go to the Window -> Preferences -> General -> Appearance.
Then I changed the "Color and Font theme" from "Default" to "Classic Theme".
Only after doing that, oomph saved my theme settings.

**Comment by CountablyInfinite** [comment]
Still not working out of the box with Neon.2 (4.6.2).
Thank you for the detailed workaround, @fschmitt

**Comment by lowley** [comment]
fixed it be choosing in preferences/general/appareance tab, and by setting Theme="Jeeeyul's themes" + oomph like said above

**Comment by max-underthesun** [comment]
for Oxygen.1a Release (4.7.1a)
no success with @fschmitt 'algorithm'
but fixed with @powergold1  'method': change 'Color and Font theme' from 'Default' to 'Classic'

**Comment by mattmilten** [comment]
I finally succeeded by using the Darkest Dark Theme. All other 'solutions' didn't do it for me. This entire issue is completely embarrassing for Eclipse.

**Comment by Catscratch** [comment]
For all using Darkest dark this thread is obsolete.

Darkest  dark is now "DevStyle". Install it from market place and all problems are gone.

Working with oxygen and win 10.

**Comment by tkitynski** [comment]
I'm experiencing the same "curse" and it is so aggravating... Spending half and hour to set up colors just to lose them all with next restart... And why on earth Oomph even tinker with colors and themes? It slows everything down, so you can sometimes observe like there's loading one (old) color scheme, then it is replaced by some colors and then again they are replaced. With all due respect, but it looks more like a hack than proper solution. And I don't mind, but if this is a hack, then there should be an easy switch to disable it.

**Comment by max-underthesun** [comment]
@tkitynski i only managed to fix this with installing Darkest Dark theme

**Comment by apteryks** [comment]
DevStyle works great, but I couldn't make it work with openjdk-jre 9 (had to install Oracle's JRE...).

**Comment by DJManas** [comment]
4.7.3 Oxygen same issue...

Ok, update.

I did what was in [this post](https://github.com/eclipse-color-theme/eclipse-color-theme/issues/223#issuecomment-118787497). But wasn’t working, then I got into settings and found out, that even when I gave recording „true“ it is still false, so re-enabled and restarted and it now seems working.
![2018-04-21_09-36-04](https://user-images.githubusercontent.com/2048276/39081801-0b73f5e6-4548-11e8-83f1-75d51ca1c772.png)

But I am wondering is Eclipse really useful if it even has problems with settings save, even after all the years it is out?

**Comment by apteryks** [comment]
Yeah, every time I use Eclipse, I embarrassingly end up spending hours configuring silly things such as colors (it's very near impossible to have a decent dark theme which is usable). DevStyle worked well, but it isn't a free (as in freedom) addition. Once the configuring done, it's quite a work horse though!

**Comment by DJManas** [comment]
Been using Eclipse, at work, since 2013 (because from some reason SAP started to build on top of it its tools). In default white mode, and after all the years I got used to it. Well I try every now and then to set colors up, but it every time ends with total reinstallation of the thing...

At home I started to use VSCode, now I tried Eclipse, just I needed to compile some code for ESP32 and didn’t want to compile whole toolchain, because the one in PlatformIO has bugs and not working at the moment... Well I guess I will end up by setting the toolchain...

**Comment by limjinsun** [comment]
<img width="691" alt="screen shot 2018-09-18 at 19 18 52" src="https://user-images.githubusercontent.com/11390856/45707935-11913a00-bb78-11e8-912f-d76d554983e1.png">
I don't know how it work from yours. but This fixed my problem.

**Comment by ewoks** [comment]
3+ years after .... :crickets:

**Comment by ctemple** [comment]
> <img alt="screen shot 2018-09-18 at 19 18 52" width="691" src="https://user-images.githubusercontent.com/11390856/45707935-11913a00-bb78-11e8-912f-d76d554983e1.png">
> 
> I don't know how it work from yours. but This fixed my problem.

it's works!

**Comment by costin3141** [comment]
oomph is the stupiest thing ever! just remove that shit. eclipse worked so well before that bs came into.

**Comment by mpages1** [comment]
It worked for me, simply deleting the .metadata folder from the workspace.

**Comment by ewoks** [comment]
anyone already preparing anniversary in May regarding #223 ? 🎂
